src/gluonts/itertools.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Iterable, Iterator, List, TypeVar
+import itertools
+import random
+
+T = TypeVar("T")
+
+
+def cycle(it):
+    """Like `itertools.cycle`, but does not store the data."""
+
+    while True:
+        yield from it
+
+
+def take(iterable: Iterable[T], n: int) -> Iterator[T]:
```

Comment:
Since we now have `toolz` as a dependency, we could just use `toolz.itertoolz.take` instead.

Reply:
nice!

Reply:
actually, what's wrong with diretly using `itertools.islice`?

Reply:
It's less descriptive. Functionally they are the same :)